internal/controller/isbservicerollout_controller_test.go

```diff
@@ -260,119 +261,133 @@ func Test_reconcile_isbservicerollout_PPND(t *testing.T) {
+			expectedRolloutPhase:       apiv1.PhaseDeployed,
+			existingInProgressStrategy: nil,
 			expectedConditionsSet: map[apiv1.ConditionType]metav1.ConditionStatus{
 				apiv1.ConditionChildResourceDeployed: metav1.ConditionTrue,
 			},
-			expectedISBSvcSpec: createDefaultISBServiceSpec("2.10.11"),
+			expectedISBSvcSpec:         createDefaultISBServiceSpec("2.10.11"),
+			expectedInProgressStrategy: apiv1.UpgradeStrategyPPND,
 		},
 		{
-			name:                   "existing ISBService - spec already updated - isbsvc reconciling",
-			newISBSvcSpec:          createDefaultISBServiceSpec("2.10.11"),
-			existingISBSvcDef:      createDefaultISBService("2.10.11", numaflowv1.ISBSvcPhaseRunning, false),
-			existingStatefulSetDef: createDefaultISBStatefulSet("2.10.3", false),
-			existingPipeline:       createDefaultPipelineOfPhase(numaflowv1.PipelinePhasePaused, map[string]string{}),
-			expectedRolloutPhase:   apiv1.PhaseDeployed,
+			name:                       "existing ISBService - spec already updated - isbsvc reconciling",
+			newISBSvcSpec:              createDefaultISBServiceSpec("2.10.11"),
+			existingISBSvcDef:          createDefaultISBService("2.10.11", numaflowv1.ISBSvcPhaseRunning, false),
+			existingStatefulSetDef:     createDefaultISBStatefulSet("2.10.3", false),
+			existingPipeline:           createDefaultPipelineOfPhase(numaflowv1.PipelinePhasePaused, map[string]string{}),
+			expectedRolloutPhase:       apiv1.PhaseDeployed,
+			existingInProgressStrategy: &ppndUpgradeStrategy,
 			expectedConditionsSet: map[apiv1.ConditionType]metav1.ConditionStatus{
 				apiv1.ConditionPausingPipelines: metav1.ConditionTrue,
 			},
-			expectedISBSvcSpec: createDefaultISBServiceSpec("2.10.11"),
+			expectedISBSvcSpec:         createDefaultISBServiceSpec("2.10.11"),
+			expectedInProgressStrategy: apiv1.UpgradeStrategyPPND,
 		},
 		{
-			name:                   "existing ISBService - spec already updated - isbsvc done reconciling",
-			newISBSvcSpec:          createDefaultISBServiceSpec("2.10.11"),
-			existingISBSvcDef:      createDefaultISBService("2.10.11", numaflowv1.ISBSvcPhaseRunning, true),
-			existingStatefulSetDef: createDefaultISBStatefulSet("2.10.11", true),
-			existingPipeline:       createDefaultPipelineOfPhase(numaflowv1.PipelinePhasePaused, map[string]string{}),
-			expectedRolloutPhase:   apiv1.PhaseDeployed,
+			name:                       "existing ISBService - spec already updated - isbsvc done reconciling",
+			newISBSvcSpec:              createDefaultISBServiceSpec("2.10.11"),
+			existingISBSvcDef:          createDefaultISBService("2.10.11", numaflowv1.ISBSvcPhaseRunning, true),
+			existingStatefulSetDef:     createDefaultISBStatefulSet("2.10.11", true),
+			existingPipeline:           createDefaultPipelineOfPhase(numaflowv1.PipelinePhasePaused, map[string]string{}),
+			expectedRolloutPhase:       apiv1.PhaseDeployed,
+			existingInProgressStrategy: nil,
```

Comment:
nit: I'm thinking in this case the incoming InProgressStrategy would probably be PPND